### PR TITLE
Add alerting component.

### PR DESCRIPTION
**Pull request by Ma-Dan**

### CODE REVIEW

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -80,3 +80,14 @@ var (
 	WorkSpaceRoles   = []string{WorkspaceAdmin, WorkspaceRegular, WorkspaceViewer}
 	SystemNamespaces = []string{KubeSphereNamespace, KubeSphereLoggingNamespace, KubeSphereMonitoringNamespace, OpenPitrixNamespace, KubeSystemNamespace, IstioNamespace, KubesphereDevOpsNamespace}
```
Comment (huojiao2006) [inline pkg/constants/constants.go]: should add alerting ns here?

Comment (Ma-Dan) [reply pkg/constants/constants.go]: Fixed

File: pkg/apis/alerting/v1alpha2/register.go [hunk 2]
```diff
@@ -0,0 +1,1380 @@
...
+	AddToContainer    = WebServiceBuilder.AddToContainer
+)
+
+func addWebService(c *restful.Container) error {
+	ws := runtime.NewWebService(GroupVersion)
+	tags := []string{"Alerting"}
+
+	//tags := []string{"ResourceType"}
+
+	ws.Route(ws.GET("/resource_type").To(alerting.DescribeResourceTypes).
```
Comment (zryfish) [inline pkg/apis/alerting/v1alpha2/register.go]: resourcetype

File: pkg/simple/client/kubernetes/client.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
```
Comment (zryfish) [inline pkg/simple/client/kubernetes/client.go]: we already have a interface to use kubernetes client, check pkg/simple/client/k8s/k8sclient.go

File: vendor/github.com/golang/protobuf/jsonpb/jsonpb.go [hunk 4]
```diff
@@ -0,0 +1,1271 @@
+// Go support for Protocol Buffers - Google's data interchange format
+//
```
Comment (zryfish) [inline vendor/github.com/golang/protobuf/jsonpb/jsonpb.go]: kubesphere is still using dep as dependency management tool, so add dependency package to Gopkg.toml, and run `dep ensure`

File: pkg/apis/alerting/v1alpha2/register.go [hunk 5]
```diff
@@ -0,0 +1,1380 @@
...
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Reads(alerting.PolicyByAlert{}).
+		Writes(alerting.ModifyPolicyByAlertResponse{}).
+		Returns(http.StatusOK, RespOK, alerting.ModifyPolicyByAlertResponse{})).
+		Consumes(restful.MIME_JSON, constants.MIME_MERGEPATCH).
+		Produces(restful.MIME_JSON)
+
+	ws.Route(ws.PATCH("/namespaces/{ns_name}/pods/{pod_name}/containers/policy").To(alerting.ModifyPolicyByAlertContainer).
+		Doc("Modify Policy By Alert Container Level").
+		Param(ws.PathParameter("ns_name", "Specify namespace").DataType("string").Required(true).DefaultValue("")).
```
Comment (zryfish) [inline pkg/apis/alerting/v1alpha2/register.go]: I think `namespace` would be a better name

File: pkg/apis/alerting/v1alpha2/register.go (hunk 2 above)
Comment (Ma-Dan) [reply pkg/apis/alerting/v1alpha2/register.go]: Fixed

File: pkg/simple/client/kubernetes/client.go (hunk 3 above)
Comment (Ma-Dan) [reply pkg/simple/client/kubernetes/client.go]: Fixed

File: vendor/github.com/golang/protobuf/jsonpb/jsonpb.go (hunk 4 above)
Comment (Ma-Dan) [reply vendor/github.com/golang/protobuf/jsonpb/jsonpb.go]: Will fix after applying go module to kubesphere project.

File: pkg/apis/alerting/v1alpha2/register.go (hunk 5 above)
Comment (Ma-Dan) [reply pkg/apis/alerting/v1alpha2/register.go]: Fixed.

**Pull request closed without merging**